ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala

```diff
@@ -59,6 +61,76 @@ sealed abstract class ContractTypeId[+PkgId]
   }
 }
 
+object ResolvedQuery {
+  def apply(resolved: ContractTypeId.Resolved): ResolvedQuery = {
+    resolved match {
+      case t: ContractTypeId.Template.Resolved => ByTemplateId(t)
+      case i: ContractTypeId.Interface.Resolved => ByInterfaceId(i)
+      case unexpected => throw new Exception(s"unexpected,  ContractTypeId.Resolved is $unexpected")
```

Comment:
The type pattern form doesn't compile, but this one does.

```suggestion
    import ContractTypeId._
    resolved match {
      case t @ Template(_, _, _) => ByTemplateId(t)
      case i @ Interface(_, _, _) => ByInterfaceId(i)
```